app/api/v1/endpoints/blog.py

```diff
@@ -75,26 +76,30 @@ def read_blogs(
     Get all blogs.
     """
 
-    redis_client = RedisClient.get_instance()
-    cashe_key = f"blogs_{page}_{limit}_{search}_{is_published}_{created_from}_{created_to}_{author_id}"
+    try:
+        redis_client = await RedisClient.get_instance()
+        cashe_key = f"blogs_{page}_{limit}_{search}_{is_published}_{created_from}_{created_to}_{author_id}"
```

Comment:
after cache created if a new blog post is created how cache will handle that scenario ?

Reply:
Using cache in this small assignment does not make sense we only use cache in reports where large processing is performed.
Still we can use count to check if posts are created but if any post got updated this scenario will get failed.
So ignore this for now.